gordo/client/cli/custom_types.py

```diff
@@ -0,0 +1,50 @@
+"""Custom click types."""
+import os
+import typing
+
+import click
+import yaml
+from dateutil import parser
+from gordo_dataset.data_provider import providers
+
+
+class DataProviderParam(click.ParamType):
+    """Load a DataProvider from JSON/YAML representation or from a JSON/YAML file."""
+
+    name = "data-provider"
+
+    def convert(self, value, param, ctx):
+        """Convert the value for data provider."""
+        if os.path.isfile(value):
```

Comment:
FYI if `value` is a YAML string, and this string contains characters unrepresentable at the OS level, `os.path.isfile` would raise an exception, and it will never get to `else`.